### PR TITLE
test: close 0.2.0 patch-coverage gaps in ingest + prune-existing

### PR DESCRIPTION
**Pull request by wjduenow**
Raises codecov patch coverage on the new 0.2.0 code (the gaps surfaced on the #111 release PR).

| File | Before | After |
|---|---|---|
| `ingest/parser.py` | 88% | **100%** |
| `ingest/reader.py` | 94% | **100%** |
| `cli/prune_existing.py` | 86% | **96%** |

**Tests added (15):**
- **parser** — `ref()`/`source()` non-quoted + arity edges, non-dict test bodies, multi-key + non-str/dict entries, model-level `relationships` skip
- **reader** — path-containment failure → `IngestSchemaParseError`, unreadable file (POSIX, non-root, perms restored), non-dict / nameless `columns:` entries ignored
- **prune-existing** — walk-up project-dir resolution (success + failure), `--no-color` env mutation, empty skipped-report guard

**Residual (documented exclusions, not real gaps):** `prune_existing.py:270` is the real `WarehouseAdapter.from_profile()` (only reachable under the `bigquery`/`e2e` gated markers — unit tests patch `_make_warehouse_adapter`); `:447-454` is the `--profiles-dir` resolve-error branch (`resolve(strict=False)` doesn't raise on ordinary input).

Full suite: 1943 passed, 96.49% coverage; ruff/format/pyright clean.

## Summary by CodeRabbit

* **Tests**
  * Expanded test coverage for CLI behavior, parser edge cases, and schema validation error handling.

[Review Change Stack](https://app.coderabbit.ai/change-stack/wjduenow/SignalForge/pull/112?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: tests/cli/test_prune_existing.py [hunk 1]
```diff
@@ -518,3 +519,121 @@ def test_bad_project_dir_exit_1(tmp_path: Path, capsys: pytest.CaptureFixture[st
...
+        "full",
+        "--sample-strategy",
+        "oneshot",
+    ]
+    factory = _make_fake_adapter_factory()
+    with patch("signalforge.cli.prune_existing._make_warehouse_adapter", factory):
+        code = main(argv)
+    assert code == 0
+    assert "Traceback" not in capsys.readouterr().err
+
```
[the remark this answers is not in the document]
Comment (wjduenow) [reply tests/cli/test_prune_existing.py:557]: Done in 9fcf551 — added `assert (project_dir / ".signalforge" / "diff.json").is_file()` to all three success-path tests (walk-up, --no-color, no-skipped). None use --dry-run, so the default-on sidecar is written.
